### ISSUE TITLE
[Feature Request]: Allow Blank Device Selection & All Buttons on AL2

### ISSUE THREAD

**Issue by NightShaman** [PRProd/HA-Firemote#305]
### Feature Request

So, 1st thing love the remote! Using it with my Shields. 

I noticed with AL2 and Overwriting buttons I could use it for my Logitech Harmony, which is AWESOME!!

What AL2 is lacking is all the buttons. I believe Fire Style 6 has them all. I really do not want to use that remote as I would be using a tab card to switch remotes depending on the activity. I prefer things look uniform and if things are too different my wife will complain. ;) 

Since I would be overwriting all the buttons anyway, I would not need to connect to the shield. Really in that room the Harmony would be able to do it all anyway. 

I figure adding Harmony might be too big of an ask. I do not know js at all, otherwise I would have given it shot.
Hope that makes sense!

**Comment by NightShaman** [comment]
Sorry just thought about it, If it would be easier to take AL1 and allow users to put in volume, channel, and pad buttons from a drop down that would work as well.

**Comment by PRProd** [comment]
Greetings @NightShaman!

I am a little confused about your request, but I'm trying to understand it.

In the title of this issue, you mentioned something about allowing a blank device selection.  While I haven't tried it, two things come to mind.  First, if you are planning on overriding 100% of the buttons anyway, it really wouldn't matter what device you choose.  The drawback to this approach is that you'd lose the "glowing button" feature that shows which apps are active.  Second, I saw an integration called "Emulated Roku".  I have almost no clue what it does, but maybe you could use that as a device placeholder?  You'll want to read the [documentation](https://www.home-assistant.io/integrations/emulated_roku) for it to find out for yourself.

As far as the AL2 releated question goes, I don't fully understand what you mean by "all the buttons", but maybe that's not important.  A feature already exists that allows you to make your own buttons that do whatever you want them to do.  Take a look at the [Launcher Button Customizations](https://github.com/PRProd/HA-Firemote?tab=readme-ov-file#launcher-button-customizations) section in the [readme](https://github.com/PRProd/HA-Firemote?tab=readme-ov-file) file, or the [Create a Custom Button for Yourself](https://github.com/PRProd/HA-Firemote/wiki#create-a-custom-button-for-yourself) section in the [Firemote wiki](https://github.com/PRProd/HA-Firemote/wik).

**Comment by NightShaman** [comment]
Sorry ADHD brain sometimes kicks in and I think everyone automaticlly knows what I am talking about. 

Correct if I override everything it doesn't matter what is chosen. It would just be nice not to have to, or lets say I didn't have a Android TV, or anything in HA but the harmony, I would have nothing to choose from to make the card work. (I hope that explains that?) The Emulated Roku may work. I was just looking at from a Harmony user perspective as the options out there are not as nice as yours. :) 

 When I say all the buttons, I mean all the buttons that can be over written. Example would be the Channel up and Down, While that is available on the fire 6 remote, it is not on the AL2. I think there are some others from the list of buttons that can be overwritten that are not displayed on the AL2. 

That was why I added if it was easier to allow users to select those buttons on the AL1 that would work as well. Kind of like a make your own remote with AL1. I know you can but I think the oval buttons you have would look nicer. Right now my Harmony one looks like this, which isn't bad, but I cannot drop a pad there easily. 
![image](https://github.com/PRProd/HA-Firemote/assets/12952292/cf18114c-c9c7-425d-9f67-8c8f1a9474a0)

I hope that makes more sense? Sorry I took a bit to reply just got back in town.

**Comment by PRProd** [comment]
@NightShaman ~

I haven't forgotten about you!  I'm experimenting with some things for the "Blank Device Selection" idea right now, and I'm hoping to have something for you to test in the near future.

-Doug

**Comment by PRProd** [comment]
@NightShaman ~

As the first step to your feature request, I had to work out the best way to allow Firemote to be OK with no entity selected at all.  I have something for you to test in beta now, and I would love your feedback.

Please download and test Firemote beta version [v3.4.0b1](https://github.com/PRProd/HA-Firemote/releases/tag/v3.4.0b1)

I'm not convinced that I have all of the bugs worked out of it yet, so please let me know if you find anything while testing this out.  I'd really appreciate it.

Thanks!
-Doug
<br>

After updating, if you are unable to see the new updates on your Firemote card, please visit [this link](https://github.com/PRProd/HA-Firemote/wiki/Force-a-Refresh) for help.
<br>

## Visual Editor:
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/4f1e7bef-b050-439f-9c70-270915fdb7f1)

## YAML config (for button overrides)
```yaml
type: custom:firemote-card
entity: none
device_family: none
device_type: other
compatibility_mode: default
button_overrides:
  volume-up-button:
    service: light.toggle
    target:
      entity_id: light.bedroom_lamp
    data:
      color_name: red
      transition: 2
      brightness_pct: 100
scale: '100'
defaultRemoteStyle_override: AL2
app_launcher_relative_size: '1'
hide_button_group_frame: true
useCustomSkin: true
dpad_style: apple-tv-black
visible_name_text: 🔥 FIREMOTE 🔥
name_position: top
```

## Firemote Card
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b42c5e92-35b1-44fd-842a-2980374e88ae)

**Comment by PRProd** [comment]
Another example:

## YAML
```yaml
type: custom:firemote-card
entity: none
device_family: none
device_type: other
compatibility_mode: default
button_overrides:
  volume-up-button:
    service: light.toggle
    target:
      entity_id: light.bedroom_lamp
    data:
      color_name: red
      transition: 2
      brightness_pct: 100
  power-button:
    hidden: true
  back-button:
    hidden: true
  home-button:
    hidden: true
  rewind-button:
    hidden: true
  playpause-button:
    hidden: true
  fastforward-button:
    hidden: true
  reboot-button:
    hidden: true
  keyboard-button:
    hidden: true
  hamburger-button:
    hidden: true
  mute-button:
    hidden: true
scale: '53'
defaultRemoteStyle_override: AL2
app_launcher_relative_size: '43'
hide_button_group_frame: true
useCustomSkin: true
dpad_style: apple-tv-black
visible_name_text: 🔥 FIREMOTE 🔥
name_position: top
custom_launchers:
  - friendly_name: flash the light
    label: FLASH
    image_path: >-
      https://upload.wikimedia.org/wikipedia/commons/2/28/Flash-outlined-thin-circular-button.svg
    color: red
    background: white
    script: flashthelablight
  - friendly_name: Red Lamp
    label: red
    icon: mdi:lamp
    color: red
    background: blue
    service: light.toggle
    target:
      entity_id: light.bedroom_lamp
    data:
      color_name: red
      transition: 2
      brightness_pct: 100
app_launch_1: customlauncher flash the light
app_launch_2: customlauncher Red Lamp
skin: '#4785b8'
```

## Firemote Card
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/7e0ab8ba-3095-488c-bd24-66da1ee07d6f)

**Comment by PRProd** [comment]
@NightShaman & @baudneo (from #361) ~

Have either of you had the opportunity to try the beta version [v3.4.0b1](https://github.com/PRProd/HA-Firemote/releases/tag/v3.4.0b1) yet?
<br>

@NightShaman ~

Does this beta and/or the examples I provided satisfy your request?  If not, can you outline what you'd like to see as the next step?

**Comment by baudneo** [comment]
I did upgrade but haven't had an opportunity to test out the new features

**Comment by baudneo** [comment]
Visual editor doesnt seem to work for me, I see no None/Other option for device family. The above example YAML does work though.

I followed all advice to refresh or remove the sticky cache without any luck. It shows the correct foremote version in dev tools.

**Comment by PRProd** [comment]
@baudneo ~

Thanks for your reply!

I sincerely appreciate your efforts to look at the dev tools and to attempt the refresh.  I'm not sure why that refresh didn't reload all of the Firemote files for you, but if I had to guess, I would say that your browser is somehow *still* holding onto the older version of the [supported-devices.js](https://github.com/PRProd/HA-Firemote/blob/v3.4.0b1/dist/supported-devices.js) file.


The only thing I can suggest is to try the refresh procedure again.  You should be seeing this:

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/8f928a4b-2da1-42ed-8853-9856a6812302)


You can double check my work too.  I have been known to make mistakes.
You can click to review the code change [in place](https://github.com/PRProd/HA-Firemote/blame/v3.4.0b1/dist/supported-devices.js#L441-L459) or in [diff](https://github.com/PRProd/HA-Firemote/compare/v3.3.3...v3.4.0b1#diff-404d875b7c3f19eb4b89d69b6a9ed9ef47d51b66c0643be62e872dab915dca2b) format
 
Please let me know what happens.  Thanks!

**Comment by PRProd** [comment]
@NightShaman & @baudneo ~

Good news!  I've released Firemote version [v3.4.0](https://github.com/PRProd/HA-Firemote/releases/tag/v3.4.0) with the ``entity: none`` feature as described above.

@baudneo ~ Thanks for confirming that the beta version worked, with only the hiccup of the config ui.  I made a little adjustment in the code structure to hopefully mitigate situations like this in the future.  Please let me know if the UI issue you found still persists.

@NightShaman ~ I'm going to close this issue as completed for now in the hopes that I've satisfied your request.  If I haven't, please feel free to reply, re-open this issue, or open a new issue to discuss layouts.

Enjoy!

-Doug

**Comment by baudneo** [comment]
Awesome! Thank you for your work and dedication!
